mdazfar2/Ezyshop
# filtering by category on shop: feature added

### **Description**  
This PR introduces **category-based filters** on the individual seller's shop page, enhancing product organization and improving the user experience by allowing customers to browse products by seller-defined categories.

I have also made the **shops page** and **individual shop products page** server components in this PR. Further increasing performance.

### **Proposed Changes**  
- **Backend:**  
  - Update the product API to **include category details** and support filtering by category.  
- **Frontend:**  
  - Add **category filter UI** on the seller’s shop page to let users filter products based on selected categories.  
  - Ensure the **UI is intuitive and consistent** with the current design.  
  - Handle **error and loading states** for a smoother user experience.

### **How to Test**  
1. Navigate to a seller’s shop page.  
2. Confirm that the **category filters** reflect the categories defined by the seller.  
3. Select a category to ensure only **products belonging to that category** are displayed.  
4. Test with sellers having **multiple categories, single category, and no categories** to ensure consistent behavior.  
5. Check the **UI responsiveness and styling** for consistency with the existing design.

### **Screenshots**  

https://github.com/user-attachments/assets/17573bbf-4d41-4150-b706-63b3f786f5a5


https://github.com/user-attachments/assets/01071452-4e2b-4cef-be78-f83d1353aca9


closes #859 
---

### **Checklist**  
- [x] Backend supports category-based product filtering.  
- [x] Frontend displays category filters and fetches products dynamically.  
- [x] UI remains consistent with the existing design.  
- [x] Error and loading states are handled properly.  

---

### **Contributions**  
- [x] I have checked the existing issues.  
- [x] I have read the [Contributing Guidelines](https://github.com/mdazfar2/Ezyshop/blob/main/CONTRIBUTING.md).  
- [x] I’m a GSSoC'24-Extd contributor.  
- [x] I’m a Hacktoberfest'24 contributor.  
- [x] I have converted the shops and individual product pages to **server components** in this PR.
